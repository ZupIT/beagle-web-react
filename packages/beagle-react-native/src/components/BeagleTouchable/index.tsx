/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC, useContext } from 'react'
import BeagleServiceContext from 'common/provider'
import { BeagleTouchableInterface } from 'common/models'
import { TouchableOpacity, StyleSheet, View } from 'react-native'


const BeagleTouchable: FC<BeagleTouchableInterface> = ({ 
  onPress,
  clickAnalyticsEvent,
  style,
  children,
}) => {
  const beagleService = useContext(BeagleServiceContext)
  const beagleAnalytics = beagleService && beagleService.analytics
  const handlePress = () => {
    if (clickAnalyticsEvent && beagleAnalytics)
      beagleAnalytics.trackEventOnClick(clickAnalyticsEvent)
    return onPress && onPress()
  }

  const styleSheet = StyleSheet.create({
    fromBffStyles: {
      ...style,
    },
    defaultStyles: {
      flex: style && style.flex ? Number(style.flex) : 1,
      justifyContent: 'center',
      alignItems: 'center',
      alignContent: 'center',
    },
    touchable:{
      flex: style && style.flex ? Number(style.flex) : 1,
    },
  })
  return (
    <TouchableOpacity onPress={handlePress} style={styleSheet.touchable}>
      <View style={{ ...styleSheet.defaultStyles, ...styleSheet.fromBffStyles }}>
        {children}
      </View>
    </TouchableOpacity>
  )
}

export default BeagleTouchable
