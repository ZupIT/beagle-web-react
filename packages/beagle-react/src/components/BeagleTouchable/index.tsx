/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC, useContext, useState } from 'react'
import BeagleServiceContext from 'common/provider'
import { BeagleTouchableInterface } from 'common/models'
import { buildAccessibility } from '../../../../common/utils/accessibility'
import withTheme from '../utils/withTheme'
import { StyledBeagleTouchable } from './styled'

const BeagleTouchable: FC<BeagleTouchableInterface> = ({ 
  onPress,
  clickAnalyticsEvent,
  className,
  style,
  children,
  accessibility,
}) => {
  const a11y = buildAccessibility(accessibility)
  const beagleService = useContext(BeagleServiceContext)
  const [pressed, setPressed] = useState(false)
  const beagleAnalytics = beagleService && beagleService.analytics
  
  const handlePress = () => {
    setPressed(false)

    if (clickAnalyticsEvent && beagleAnalytics)
      beagleAnalytics.trackEventOnClick(clickAnalyticsEvent)
    return onPress && onPress()
  }
  
  return (
    <StyledBeagleTouchable 
      className={className}
      onMouseDown={e => setPressed(true)} 
      onClick={handlePress} 
      style={style}
      aria-pressed={pressed}
      {...a11y}>
      {children}
    </StyledBeagleTouchable>
  )
}

export default withTheme(BeagleTouchable)
