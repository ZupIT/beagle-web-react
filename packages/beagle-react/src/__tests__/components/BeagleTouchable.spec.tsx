/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

// Link.react.test.js
import React from 'react'
import 'jest-styled-components'
import Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import BeagleTouchable from '../../components/BeagleTouchable'

let wrapper: any
const onPressMock = jest.fn()

configure({ adapter: new Adapter() })
beforeAll(() => {
  wrapper = shallow(
    <BeagleTouchable onPress={onPressMock}>
      <p>Important Content</p>
    </BeagleTouchable>
  )
})

test('Beagle snapshot Touchable', () => {
  expect(wrapper).toMatchSnapshot()
})

test('Should call on click handler', () => {
  wrapper.simulate('press')
  expect(onPressMock).toHaveBeenCalledTimes(1)
})
