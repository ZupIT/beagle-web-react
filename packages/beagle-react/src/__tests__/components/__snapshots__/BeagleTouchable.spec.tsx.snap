// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beagle snapshot Touchable 1`] = `
<BeagleTouchable
  className=""
  onPress={[MockFunction]}
>
  <p>
    Important Content
  </p>
</BeagleTouchable>
`;
